bionty/dev/_handle_versions.py

```diff
@@ -1,163 +1,192 @@
-    )
-    defaults_yaml = (
-        load_yaml(_CURRENT_PATH) if defaults == "current" else load_yaml(_LNDB_PATH)
-    )
+    if not CURRENT_VERSIONS_PATH.exists() or overwrite:
+        source_path = (
+            PUBLIC_VERSIONS_PATH if source == "versions" else LOCAL_VERSIONS_PATH
+        )
 
-    missing_defaults = []
-    missing_entites = set(versions_yaml.keys()) - set(defaults_yaml.keys())
-    missing_entites.remove("version")
+        write_yaml(parse_current_versions(source_path), CURRENT_VERSIONS_PATH)
 
-    for entity in missing_entites:
-        entity_content = list(versions_yaml.get(entity).items())
-        for content in entity_content:
-            database = content[0]
-            versions_species = content[1]
-            species: list[str] = versions_species.get("species", {})
-            latest_version = list(versions_species.get("versions", {}).keys())[0]
 
-            for spec in species:
-                missing_defaults.append((entity, database, spec, latest_version))
+def update_local_from_versions_yaml():
+    """Update LOCAL_VERSIONS_PATH to add additional entries from PUBLIC_VERSIONS_PATH."""
+    public_df_records = parse_versions_yaml(PUBLIC_VERSIONS_PATH).to_dict(
+        orient="records"
+    )
+    local_df_records = parse_versions_yaml(LOCAL_VERSIONS_PATH).to_dict(
+        orient="records"
+    )
+    additional_records = [i for i in public_df_records if i not in local_df_records]
+    if len(additional_records) > 0:
+        updated_local_versions = add_records_to_versions_yaml(
+            additional_records, LOCAL_VERSIONS_PATH
+        )
+        write_yaml(updated_local_versions, LOCAL_VERSIONS_PATH)
+        logger.success(
+            "New records found in the public version.yaml, updated"
+            f" {LOCAL_VERSIONS_PATH}!"
+        )
+        # update LOCAL_VERSIONS_PATH will always generate new CURRENT_VERSIONS_PATH
+        create_current_versions_yaml(overwrite=True)
+
+
+def parse_current_versions(yamlpath: Union[str, Path]):
+    """Parse out the most recent versions from yaml."""
+    df = parse_versions_yaml(yamlpath)
+    df_current = (
+        df[["entity", "source_key", "species", "version"]]
+        .drop_duplicates(["entity", "species", "source_key"], keep="first")
+        .groupby(["entity", "species", "source_key"], sort=False)
+        .max()
+    )
 
-    return missing_defaults
+    current_dict: Dict = {}
+    for kwargs in df_current.reset_index().to_dict(orient="records"):
+        entity, species, source_key, version = (
+            kwargs["entity"],
+            kwargs["species"],
+            kwargs["source_key"],
+            kwargs["version"],
+        )
+        if entity not in current_dict:
+            current_dict[entity] = {}
+        if species not in current_dict[entity]:
+            current_dict[entity][species] = {source_key: version}
+    return current_dict
+
+
+def add_records_to_versions_yaml(records: List[dict], yaml_filepath):
```

Comment:
```suggestion
def add_records_to_versions_yaml(records: List[Dict], yaml_filepath):
```

Our users still want support for older Python versions